File: DroidsAPI/src/main/java/com/github/rochedo098/droidsapi/api/events/ModuleEventListener.java
package com.github.rochedo098.droidsapi.api.events;

import net.minecraft.entity.Entity;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Experimental
public interface ModuleEventListener {
    default void onApply(Entity entity) {}

    default void onRemove(Entity entity) {}

    default void onBreak() {}
}

File: src/main/kotlin/com/github/rochedo098/droids/api/events/ModuleEventListener.kt
package com.github.rochedo098.droids.api.events

import net.minecraft.entity.Entity
import org.jetbrains.annotations.ApiStatus

@ApiStatus.Experimental
interface ModuleEventListener {
    fun onApply(entity: Entity?) {}

    fun onRemove(entity: Entity?) {}

    fun onBreak() {}
}